llmproxy/llmproxy.py

```diff
@@ -126,6 +126,33 @@ def _setup_user_models(available_models=None, settings=None) -> Dict[str, object
         raise UserConfigError(f"Unknown error occured during llmproxy.config setup:{e}")
 
 
+def load_model_costs(path_to_yml: str, provider_name: str) -> dict:
```

Comment:
This function is unnecessary. We can do all of this setup for category and cost within the initial _setup_available_models. This will allow us to avoid the overhead of re-reading the yml files. We want to avoid reading YAML files as it is an expensive operation

Reply:
You can instead feed in the configuration needed through the get_completion() function or some other method that avoid reading the YAMl file multiple times.